educator/tests/Test/Dscp/Educator/Mode.hs

```diff
@@ -111,9 +112,13 @@ runTestSqlM testDb action =
         let _tecAppDir = error "AppDir is not defined"
         let _tecIssuerInfo = KnownIssuerInfo certificateIssuerInfoEx
         let ctx = TestEducatorCtx{..}
+        let workers = [txRetranslatingWorker]
+
         runRIO ctx $ markWithinWriteSDLockUnsafe applyGenesisBlock
 
-        liftIO . rollbackInEnd $ runRIO ctx action
+        liftIO . rollbackInEnd $ runRIO ctx $
+            withWorkers (pure workers) $
+                action
```

Comment:
> Btw, why do we even call txPublisher listener? In my world, listeners are threads which listen for messages from outside world

As far as I understand @volhovm's architecture, anything related to the client is worker while threads acting as servers are listeners. And `txPublisher` is a server since it submits transactions over `PUB` socket.

Reply:
Correct, listeners are server components, and workers are client components. Clients can not "publish" because they don't have PUB sockets, but listeners can.